packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -2098,34 +2098,21 @@ addEventListener('fetch', event => {});`
       mockSubDomainRequest();
       mockListKVNamespacesRequest(kvNamespace);
       mockKeyListRequest(kvNamespace.id, []);
+      const requests = mockUploadAssetsToKVRequest(kvNamespace.id);
+
+      await runWrangler("publish");
 
       // We expect this to be uploaded in 3 batches
       // The first batch has 11 files (88mb)
       // The next batch has 5 files (93mb)
       // And the last one has 4 files (98mb)
-
-      // Since the bulk upload api endpoint stays the same
-      // We're going to have to clear the mock as soon as it's resolved
-      // And immediately add a mock for another one
-      // Welcome to a callback pyramid in 2022
-
-      const removeFirstBulkUploadMockFn = mockUploadAssetsToKVRequest(
-        kvNamespace.id,
-        assets.slice(0, 11),
-        () => {
-          removeFirstBulkUploadMockFn();
-          const removeSecondBulkUploadMockFn = mockUploadAssetsToKVRequest(
-            kvNamespace.id,
-            assets.slice(11, 16),
-            () => {
-              removeSecondBulkUploadMockFn();
-              mockUploadAssetsToKVRequest(kvNamespace.id, assets.slice(16, 20));
-            }
-          );
+      let assetIndex = 0;
```

Comment:
This test also checks that the batching happens correctly, (11, 5, 4) wouldn't this rewrite lose that check?

Reply:
Oh, I see. Right. This now tests that the uploaded items appear in the correct order, but not the size of each batch. I will add that in now.